ui/src/components/shared/TableCard.tsx

```diff
@@ -18,16 +18,16 @@ export function TableCard({
   defaultContents: ReactNode;
   extraButton?: ReactNode;
   isLoading: boolean;
-  onSearch: (value: string) => void;
-  query: string | null;
-  searchPlaceholder: string;
+  onSearch?: (value: string) => void;
+  query?: string | null;
+  searchPlaceholder?: string;
   showDefaultContents: boolean;
   table: ReactNode;
   title: ReactNode;
 }) {
   return (
     <Card bodyStyle={{ padding: 0 }} title={title}>
-      {!showDefaultContents && (
+      {!showDefaultContents && onSearch && searchPlaceholder && query !== undefined && (
```

Comment:
If `query` is defaulted to `null`, then checking for `query !== undefined` doesn't make sense?